tests/msc2716_test.go

```diff
@@ -536,30 +581,16 @@ func TestBackfillingHistory(t *testing.T) {
 				},
 			})
 
-			// Send a marker event to let all of the homeservers know about the
-			// insertion point where all of the historical messages are at
-			markerEvent := b.Event{
-				Type: markerEventType,
-				Content: map[string]interface{}{
-					markerInsertionContentField: baseInsertionEventID,
-				},
-			}
-			// We can't use as.SendEventSynced(...) because application services can't use the /sync API
-			markerSendRes := as.MustDoFunc(t, "PUT", []string{"_matrix", "client", "r0", "rooms", roomID, "send", markerEvent.Type, "txn-m123"}, client.WithJSONBody(t, markerEvent.Content))
-			markerSendBody := client.ParseJSON(t, markerSendRes)
-			markerEventID := client.GetJSONFieldStr(t, markerSendBody, "event_id")
-
-			// Make sure the marker event has reached the remote homeserver
-			remoteCharlie.SyncUntilTimelineHas(t, roomID, func(ev gjson.Result) bool {
-				return ev.Get("event_id").Str == markerEventID
-			})
+			// Send the marker event
+			sendMarkerAndEnsureBackfilled(t, as, remoteCharlie, roomID, baseInsertionEventID)
```

Comment:
Split all of this sending and verifying logic into its own function `sendMarkerAndEnsureBackfilled` which we also re-use 🎉